src/Lykke.Job.ChainalysisHistoryExporter/Deposits/DepositsHistoryProviders/Ripple/XrpDepositsHistoryProvider.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Flurl.Http;
+using Lykke.Job.ChainalysisHistoryExporter.Common;
+using Lykke.Job.ChainalysisHistoryExporter.Reporting;
+using Lykke.Job.ChainalysisHistoryExporter.Settings;
+
+namespace Lykke.Job.ChainalysisHistoryExporter.Deposits.DepositsHistoryProviders.Ripple
+{
+    public class XrpDepositsHistoryProvider : IDepositsHistoryProvider
+    {
+        private readonly Blockchain _ripple;
+        private readonly XrpSettings _settings;
+
+        public XrpDepositsHistoryProvider(
+            IBlockchainsProvider blockchainsProvider,
+            XrpSettings settings)
+        {
+            _ripple = blockchainsProvider.GetRipple();
+            _settings = settings;
+        }
+
+        public bool CanProvideHistoryFor(DepositWallet depositWallet)
+        {
+            return depositWallet.CryptoCurrency == _ripple.CryptoCurrency;
+        }
+
+        public async Task<PaginatedList<Transaction>> GetHistoryAsync(DepositWallet depositWallet, string continuation)
+        {
+            if (!CanProvideHistoryFor(depositWallet))
+            {
+                return PaginatedList.From(Array.Empty<Transaction>());
+            }
+
+            if (!long.TryParse(continuation, out var ledgerIndexMin))
+            {
+                ledgerIndexMin = -1;
+            }
+
+            var addressParts = depositWallet.Address.Split('+', StringSplitOptions.RemoveEmptyEntries);
+            var address = addressParts[0];
+            var tag = addressParts.Length > 1
+                ? addressParts[1]
+                : null;
+
+            var transactions = new List<Transaction>();
+            object pagingMarker = null;
+
+            do
+            {
+                var response = await _settings.RpcUrl
```

Comment:
Would be nice to have a cache for this, since actually we'll get the history of the same address as many time as many wallets associated with it. The cache can be invalidated after the single job run or after a configurable time.